credentials/credentials_test.go

```diff
@@ -58,4 +62,150 @@ func TestTLSClone(t *testing.T) {
 	if c.Info().ServerName != expectedServerName {
 		t.Fatalf("Change in clone should not affect the original, c.Info().ServerName = %v, want %v", c.Info().ServerName, expectedServerName)
 	}
+
+}
+
+const tlsDir = "../test/testdata/"
+
+type serverHandshake func(net.Conn) (tls.ConnectionState, error)
```

Comment:
can we define it as security-protocol agnostic -- it should return AuthInfo instead of tls.ConnectionState?